pkg/datapath/loader/loader.go

```diff
@@ -360,7 +362,7 @@ func (l *Loader) reloadDatapath(ctx context.Context, ep datapath.Endpoint, dirs
 				scopedLog.WithError(err).Warn("Failed to upsert route")
 			}
 		} else {
-			if err := removeEndpointRoute(ep, *ip.IPNet(32)); err != nil {
+			if err := removeEndpointRoute(ep, *ip.IPNet(32)); err != nil && !errors.Is(err, syscall.ESRCH) {
```

Comment:
Is that returned the route was already removed? If so, when and why was it removed?

Reply:
Discussed offline: given it's called in `reloadDatapath`, we will attempt to delete the route every time we reload the endpoint's BPF programs, even if we never even installed the route. So the ESRCH makes sense :+1: